src/signing/rsa-signature.ts

```diff
@@ -0,0 +1,47 @@
+import { pki, md } from 'node-forge';
+import { encodeSafe64, decodeSafe64 } from '../../src/util';
+
+
+export interface ISignature {
+    signature: string;
+    data: string;
+}
+
+export function signWithPrivateKey(privateKeyPem: string, data: string): ISignature {
+    const mdDigest = md.sha256.create();
+    const key = pki.privateKeyFromPem(privateKeyPem) as pki.rsa.PrivateKey;
+    mdDigest.update(data, 'utf8');
+    const signature = key.sign(mdDigest);
+    return {
+        signature: signature,
+        data: data
+    };
+}
+
+export function serializeRsaSignature(signatureObj: ISignature): string{
+    return `Sign.Rsa4096.${encodeSafe64(signatureObj.signature)}.${encodeSafe64(signatureObj.data)}`;
+}
+
+export function loadRsaSignature(serializedPayload: string): ISignature{
+    const decomposedPayload = serializedPayload.split('.')
+    const signed = decomposedPayload[0]
+    const signingStrategy = decomposedPayload[1]
+    const encodedSignature = decomposedPayload[2]
+    const encodedData = decomposedPayload[3]
```

Comment:
Use array destructuring for these.

Reply:
[signed, signingStrategy, encodedSignature, encodedData] = decomposedPayload